app/plugin.go

```diff
@@ -400,6 +402,96 @@ func (a *App) GetPlugins() (*model.PluginsResponse, *model.AppError) {
 	return resp, nil
 }
 
+// GetMarketplacePlugins returns a list of plugins from the marketplace-server,
+// and plugins that are installed locally.
+func (a *App) GetMarketplacePlugins(filter *model.MarketplacePluginFilter) ([]*model.MarketplacePlugin, *model.AppError) {
+	var result []*model.MarketplacePlugin
+	pluginSet := map[string]bool{}
+	pluginsEnvironment := a.GetPluginsEnvironment()
+	if pluginsEnvironment == nil {
+		return nil, model.NewAppError("GetMarketplacePlugins", "app.plugin.config.app_error", nil, "", http.StatusInternalServerError)
+	}
+
+	marketplaceClient, err := marketplace.NewClient(
+		*a.Config().PluginSettings.MarketplaceUrl,
+		a.HTTPService,
+	)
+	if err != nil {
+		return nil, model.NewAppError("GetMarketplacePlugins", "app.plugin.marketplace_client.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	marketplacePlugins, err := marketplaceClient.GetPlugins(filter)
+	if err != nil {
+		return nil, model.NewAppError("GetMarketplacePlugins", "app.plugin.marketplace_plugins.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	for _, plugin := range marketplacePlugins {
+		installedVersion := ""
+		if pluginsEnvironment != nil {
+			var manifest *model.Manifest
+			if manifest, err = pluginsEnvironment.Manifest(plugin.Manifest.Id); err == nil {
+				// Plugin is installed, overwrite marketplace manifest.
```

Comment:
Why do we override the manifest? Presumably this also clobbers the available version, preventing a future `Upgrade` prompt?

Reply:
Correct, we did this on purpose to override any upgrade functionality as webclient wasn't ready to handle them. However, this was before when we had `Installed: (true|false)`, but with the new `InstalledVersion` field, we don't have to override, we can just send the latest manifest. 

Will fix it to not override.